File: 944.删列造序.go
/*
 * @lc app=leetcode.cn id=944 lang=golang
 *
 * [944] 删列造序
 */

// @lc code=start
func minDeletionSize(strs []string) (ans int) {
	for j := range strs[0] {
		for i := 1; i < len(strs); i++ {
			if strs[i-1][j] > strs[i][j] {
				ans++
				break
			}
		}
	}
	return
}

// @lc code=end

File: 944.删列造序.py
#
# @lc app=leetcode.cn id=944 lang=python3
#
# [944] 删列造序
#

# @lc code=start
class Solution:
    def minDeletionSize(self, strs: List[str]) -> int:
        if not strs:
            return 0
        m, n = len(strs), len(strs[0])
        res = 0
        for i in range(n):
            for j in range(1, m):
                if strs[j][i] < strs[j - 1][i]:
                    res += 1
                    break
        return res
# @lc code=end

